app/services/manual_runner.rb

```diff
@@ -0,0 +1,46 @@
+class ManualRunner
+  def initialize(repo)
+    raise "Repo doesn't exist" unless Repo.exists?(repo.id)
+    @repo = repo
+    @octokit = Octokit::Client.new(access_token: Rails.application.secrets.github_api_token)
+  end
+
+  def run_last(commits_count)
+    run_paginated(commits_count)
+  end
+
+  private
+
+  def run_paginated(commits_count, page = 1)
+    unless (commits_count <= 0)
+      count_run = run_commits(page)
+      run_paginated(commits_count - count_run, page + 1)
+    end
+  end
+
+  def run_commits(page)
+    fetched_commits = @octokit.commits("#{@repo.organization.name}/#{@repo.name}", per_page: 100, page: page)
+    formatted_commits = format_commits(fetched_commits)
+    CommitsRunner.run(formatted_commits)
+
+    fetched_commits.count
+  end
+
+  def format_commits(commits)
+    commits.map do |commit|
+      {
+        sha: commit['sha'],
+        message: commit['commit']['message'],
+        repo: {
+          id: @repo.id,
+          name: @repo.name
+        },
+        url: commit['html_url'],
+        created_at: commit['commit']['author']['date'],
+        author: {
+          name: commit['commit']['author']['name']
+        }
+      }
```

Comment:
Thank you guys for reviews ! :bowing_man: 

> not sure it's worth allocating a new hash


Excuse me for no explanation why do I format commits before passing them to `CommitsRunner`.

It is simply because commits fetched from api and webhook don't have the same format and `CommitsRunner` is intended to handle both manually and automatically (via webhook) fetched commits.

So far, we've only handled commits from webhook via [`GithubEventHandler`](https://github.com/ruby-bench/ruby-bench-web/blob/master/app/services/github_event_handler.rb). Since now we have introduced `ManualRunner` I want to move creating and running commits in `CommitsRunner`. Therefore `GithubEventHandler` and `ManualRunner` should only receive commits and pass them to `CommitsRunner` in specific format.

> Since :repo and :author references a hash as well, it would rather be 6000 times creating a Hash

I wasn't thinking about performance too much here simply because I don't find it very crucial in this scenario as @noahgibbs noticed. But creating 2000 hashes instead of 6000 can be fixed with this:
```
{
  sha: commit['sha'],
  message: commit['commit']['message'],
  repo_id: @repo.id,
  repo_name: @repo.name,
  url: commit['html_url'],
  created_at: commit['commit']['author']['date'],
  author_name: commit['commit']['author']['name']
}
```
I'll change that :+1: